### PR TITLE
chore: Internal Menu interface for ProLayout usage

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [x] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Internal API update. Do not record in changelog.       |
| 🇨🇳 Chinese |  -        |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/menu/MenuContext.tsx [hunk 1]
```diff
@@ -9,6 +9,8 @@ export interface MenuContextProps {
   antdMenuTheme?: MenuTheme;
   direction?: DirectionType;
   firstLevel: boolean;
+  /** @private Internal Usage. Safe to remove */
+  disableTitleTooltip?: boolean;
```
Comment (chenshuai2144) [inline components/menu/MenuContext.tsx:13]:
```suggestion
  disableMenuItemTitleTooltip?: boolean;
```

Comment (afc163) [reply components/menu/MenuContext.tsx:13]: showMenuItemTooltip ？

Comment (vagusX) [reply components/menu/MenuContext.tsx:13]: ts 中有个 `@internal` 注释，考虑用一下嘛

Comment (vagusX) [reply components/menu/MenuContext.tsx:13]: 如果不是对外暴露的话，可以在 tsconfig 里面开启 `stripInternal` 把 internal 的值做编译擦除

Comment (zombieJ) [reply components/menu/MenuContext.tsx:13]: `@private` 够了，万一 Pro 里面需要用到擦除了反而会报错了。

**Review by chenshuai2144 — APPROVED** [review]

**Pull request merged**